src/modules/context2d.js

```diff
@@ -2007,28 +2034,34 @@ import { console } from "../libs/console.js";
     sortPages(pages);
 
     var clipPath, oldSize, oldLineWidth;
-    if (this.autoPaging === true) {
+    if (this.autoPaging) {
       var min = pages[0];
       var max = pages[pages.length - 1];
       for (var i = min; i < max + 1; i++) {
         this.pdf.setPage(i);
 
+        var topMargin = (i === min ? this.posY + this.margin[0] : this.margin[0]);
+        var firstPageHeight = this.pdf.internal.pageSize.height - this.posY - this.margin[0] - this.margin[2];
+        var pageHeightMinusMargin = this.pdf.internal.pageSize.height - this.margin[0] - this.margin[2];
+        var pageWidthMinusMargin = this.pdf.internal.pageSize.width - this.margin[1];
+        var previousPageHeightSum = i === 1 ? 0 : firstPageHeight + (i - 2) * pageHeightMinusMargin;
+
         if (this.ctx.clip_path.length !== 0) {
           var tmpPaths = this.path;
           clipPath = JSON.parse(JSON.stringify(this.ctx.clip_path));
           this.path = pathPositionRedo(
             clipPath,
-            this.posX,
-            -1 * this.pdf.internal.pageSize.height * (i - 1) + this.posY
+            this.posX + this.margin[3],
+            -1 * previousPageHeightSum + topMargin + this.ctx.prevPageLastElemOffset
```

Comment:
prevPageLastElemOffset is not needed because getPagesByPath is already returning the correct page the character should be on.
```suggestion
            -1 * previousPageHeightSum + topMargin
```